src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/SemanticTokensCache.cs

```diff
@@ -94,7 +96,8 @@ internal sealed class SemanticTokensCache
             DocumentUri uri,
             VersionStamp semanticVersion,
             Range requestedRange,
-            [NotNullWhen(true)] out (Range Range, ImmutableArray<int> Tokens)? cachedTokens)
+            [NotNullWhen(true)] out (Range Range, ImmutableArray<int> Tokens)? cachedTokens,
+            ILogger? logger = null)
```

Comment:
Do we have to have a default for the logger? I don't like out parameters not being last 😛

Reply:
Totally fair! I'll update to make it a required param